services/src/main/java/org/fao/geonet/api/tools/i18n/TranslationPackBuilder.java

```diff
@@ -0,0 +1,365 @@
+import jeeves.server.context.ServiceContext;
+import org.apache.commons.lang.StringUtils;
+import org.fao.geonet.api.standards.StandardsUtils;
+import org.fao.geonet.domain.*;
+import org.fao.geonet.kernel.GeonetworkDataDirectory;
+import org.fao.geonet.kernel.SchemaManager;
+import org.fao.geonet.repository.*;
+import org.fao.geonet.util.XslUtil;
+import org.jdom.Element;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.Resource;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+@Service
+public class TranslationPackBuilder {
+    @Autowired
+    TranslationPackBuilder self; // For cache to work.
+
+    @Autowired
+    GeonetworkDataDirectory dataDirectory;
+    @Autowired
+    StatusValueRepository statusValueRepository;
+    @Autowired
+    MetadataCategoryRepository categoryRepository;
+    @Autowired
+    GroupRepository groupRepository;
+    @Autowired
+    OperationRepository operationRepository;
+    @Autowired
+    SourceRepository sourceRepository;
+    @Autowired
+    SchematronRepository schematronRepository;
+    @Autowired
+    IsoLanguageRepository isoLanguageRepository;
+    @Autowired
+    TranslationsRepository translationsRepository;
+    @Autowired
+    SchemaManager schemaManager;
+
+    Path jsonLocaleDirectory;
+
+    @Resource(name = "translationPacks")
+    private Map<String, List<String>> packages;
+
+    private static final String TYPE_SEPARATOR = "/";
+    private static final String LIST_SEPARATOR = "\\+";
+
+    private static final List<String> TRANSLATION_TABLES =
+        Arrays.asList(new String[]{
+        "StatusValue", "MetadataCategory", "Group", "Operation",
+        "Source", "Schematron", "IsoLanguage", "Translations"
+    });
+
+    public TranslationPackBuilder() {
+    }
+
+    public Map<String, List<String>> getPackages() {
+        return packages;
+    }
+
+    public void setPackages(Map<String, List<String>> packages) {
+        this.packages = packages;
+    }
+
+    @PostConstruct
+    public void init() {
+        jsonLocaleDirectory =
+            dataDirectory.getWebappDir().endsWith("src/main/webapp")
+                // Dev mode
+                ? dataDirectory.getWebappDir()
+                // web/src/main/webapp
+                .getParent().getParent().getParent().getParent()
+                .resolve("web-ui/src/main/resources/catalog/locales")
+                : dataDirectory.getWebappDir().resolve("catalog").resolve("locales");
+    }
+
+    @Cacheable(
+        value = "translations",
+        cacheManager = "cacheManager",
+        key = "{#schema, #key}")
```

Comment:
The key I think should have the language, otherwise translations are always returned in English. What is the `#schema` value?